provider/core/src/data_provider.rs

```diff
@@ -74,6 +74,23 @@ where
     }
 }
 
+/// A data provider that can determine whehter it can load a particular data identifier,
+/// potentially cheaper than performing the load.
+pub trait CanLoad<M: DataMarker>: DataProvider<M> {
+    /// This method returns false iff [`load`] fails with a [`DataErrorKind::MissingLocale`].
+    ///
+    /// Other errors are returned as [`load`] would.
+    ///
+    /// [`load`]: DataProvider::load
+    fn can_load(&self, req: DataRequest) -> Result<bool, DataError> {
+        match self.load(req) {
+            Ok(_) => Ok(true),
+            Err(e) if e.kind == DataErrorKind::MissingLocale => Ok(false),
+            Err(e) => Err(e),
+        }
```

Comment:
Hmmm, I'm worried that returning only `bool` doesn't differentiate between directly supported locales and supported locales by fallback. What about changing the return value to an enum?
```
enum IsLocaleLoadableBikeshed {
    Yes,
    WithFallback,
    No
}
```

Reply:
Fallback loading is not really a concept in this crate, it's something the `LocaleFallbackAdapter` does. My plan is that `LocaleFallbackAdapter` extends on this with `can_load_with_fallback` and `can_load_with_fallback_no_root`.

Reply:
What about internal fallback though? Using `LocaleFallbackAdapter` would probably only work with external fallback, IIRC.

Reply:
The baked provider would get these methods as well

Reply:
So would there be a `CanLoadWithoutFallback` trait then? Or how would you accept both `LocaleFallbackAdapter`-based providers and baked providers using internal fallback in a generic way?

Reply:
How about this?

Reply:
Yep, returning the metadata works for me!